File: android/src/main/java/com/bitmovin/player/reactnative/NowPlayingModule.kt
package com.bitmovin.player.reactnative

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NowPlayingModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("NowPlayingModule")

        AsyncFunction("isEnabled") { _: String ->
            // The Now Playing integration is iOS-only, always disabled on Android.
            false
        }

        AsyncFunction("setEnabled") { _: String, _: Boolean ->
            // The Now Playing integration is iOS-only, no-op on Android.
        }
    }
}

File: src/modules/NowPlayingModule.ts
import { NativeModule, requireNativeModule } from 'expo-modules-core';

export type NowPlayingModuleEvents = Record<string, any>;

declare class NowPlayingModule extends NativeModule<NowPlayingModuleEvents> {
  /**
   * Get whether the Now Playing integration is enabled for the specified player.
   */
  isEnabled(playerId: string): Promise<boolean | null>;

  /**
   * Enable or disable the Now Playing integration for the specified player.
   */
  setEnabled(playerId: string, enabled: boolean): Promise<void>;
}

export default requireNativeModule<NowPlayingModule>('NowPlayingModule');
